docs: update README with `atlas --help`

OmarIthawi opened the pull request:
This helps engineers to understand the project at a glance ~even though README can be outdated a bit~ while ensuring the section is up to date.

Yes, now `atlas --help` prints reStructredText (RST) but hopefully it's not too intrusive.

This contribution is part of the [FC-0012 project](https://openedx.atlassian.net/l/cp/XGS0iCcQ) which is sparked by the [Translation Infrastructure update OEP-58](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0058-arch-translations-management.html#specification).

OmarIthawi commented:
> I assume these changes won't be a problem when running it, but I just want to verify you've run `gengetoptions`
> 
> https://github.com/openedx/openedx-atlas/blob/47a9eb8d239b2f83b45d9b20b21decceaa13fc73/Makefile#L7-L10
> 
> and everything still looks correct

Yes @brian-smith-tcril, actually the CI fails everytime that I forget to run `gengetoptions`. I see a test case for it and it seems to be working properly.

OmarIthawi commented on README.rst:
```diff
@@ -34,27 +34,73 @@ Installing Locally
 
+This help message below is copied from both ``atlas --help`` and ``atlas pull --help``.
+
+It's meant to be outdated but still useful if the CLI is not installed locally.
```
Thanks. I've added both test case and `make atlas_help_to_readme` to take care of this.

OmarIthawi reviewed (commented):
added automation for the `--help` section

openedx-webhooks commented:
Thanks for the pull request, @OmarIthawi!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

brian-smith-tcril commented on atlas:
```diff
@@ -155,63 +150,59 @@ parse() {
-       The same filter is applied to all DIRECTORY MAPPINGS arguments.
+* ``-f`` or ``--filter``: A comma-separated (or space-separated) list of patterns match files and sub-directories. This is mainly useful to filter specific languages to download.
 
-          --filter=fr_CA,ar,es_419 will match both directories named 'es_419' and files named 'es_419.json'
-                                   among others
+The same filter is applied to all DIRECTORY MAPPINGS arguments for example ``--filter=fr_CA,ar,es_419`` will match both directories named 'es_419' and files named 'es_419.json' among others.
```
These lines are really long and don't wrap nicely in the terminal, it's probably worth doing some manual line breaks

OmarIthawi replied:
I'll see if this is possible without affecting `atlas --help` or the README.rst negatively.

OmarIthawi replied:
This should be resolved now.

brian-smith-tcril commented on atlas:
```diff
@@ -155,63 +150,59 @@ parse() {
+The same filter is applied to all DIRECTORY MAPPINGS arguments for example ``--filter=fr_CA,ar,es_419`` will match both directories named 'es_419' and files named 'es_419.json' among others.
 
-Example:
+Example::
```
```suggestion
Example:
```

OmarIthawi replied:
This is intentional to create code blocks on RST. Yes, it's a big of a hassle, unless we decide to embed the `atlas --help` as a big code block in README.rst which makes life much easier imho.

brian-smith-tcril replied:
That feels like a reasonable solution to me! That way we can focus on having the formatting look good in the terminal and it'll translate nicely into the RST.

brian-smith-tcril commented on atlas:
```diff
@@ -155,63 +150,59 @@ parse() {
             translations/frontend-component-header/src/i18n/messages:frontend-component-header
 
-Will result in the following tree:
+Will result in the following tree::
```
```suggestion
Will result in the following tree:
```

brian-smith-tcril commented on atlas:
```diff
@@ -222,7 +213,6 @@ Will result in the following tree:
         ├── es_419.json
         └── fr_CA.json
 
-More examples are available in the repository docs: https://github.com/openedx/openedx-translations .
 
 
 
```
I know this isn't a change from this PR, but it'd probably be good to remove a couple of the newlines here

OmarIthawi replied:
The root-cause is `gengetoptions`. I've removed one new line, but the rest are enforced by `gengetoptions`.

brian-smith-tcril commented on atlas:
```diff
@@ -155,63 +150,59 @@ parse() {
 
-    `-b` or `--branch`:
-        The git branch defaults to 'main'.
+* ``-r`` or ``--repository``: A repository GitHub repo slug defaults to 'openedx/openedx-translations'.
```
```suggestion
* ``-r`` or ``--repository``: slug of the GitHub repository to pull from. Defaults to 'openedx/openedx-translations'.
```

brian-smith-tcril commented on atlas:
```diff
@@ -155,63 +150,59 @@ parse() {
-    `-f` or `--filter`
-       A comma-separated (or space-separated) list of patterns match files and sub-directories.
-       This is mainly useful to filter specific languages to download.
+* ``-b`` or ``--branch``: The git branch defaults to 'main'.
```
```suggestion
* ``-b`` or ``--branch``: Branch to pull from. Defaults to 'main'.
```

brian-smith-tcril commented on atlas:
```diff
@@ -155,63 +150,59 @@ parse() {
+The second directory (TO_DIR) represents a local directory to copy files to.
 
-   At least one directory pair is required:
+At least one directory pair is required::
```
```suggestion
At least one directory pair is required:
```

brian-smith-tcril commented on atlas:
```diff
@@ -155,63 +150,59 @@ parse() {
-       A comma-separated (or space-separated) list of patterns match files and sub-directories.
-       This is mainly useful to filter specific languages to download.
+* ``-b`` or ``--branch``: The git branch defaults to 'main'.
 
```
Do we need these newlines in-between the bullet points? It looks quite odd in the terminal.

OmarIthawi replied:
I'll check them.

OmarIthawi replied:
I've reverted it back to the previous format. Now this shouldn't be a problem.

OmarIthawi commented:
@brian-smith-tcril I've updated it to include `atlas --help` as a code block and removed the rst-hacks from atlas so it should be more terminal-friendly.

The preview for the README is available here: https://github.com/Zeit-Labs/openedx-atlas/blob/[readme-doc](https://github.com/Zeit-Labs/openedx-atlas/blob/readme-docs/README.rst)s/README.rst

openedx-webhooks commented:
@OmarIthawi 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-atlas&entry.867055334=https://github.com/openedx/openedx-atlas/pull/28&entry.1484655318=https://github.com/OmarIthawi&entry.752974735=2023-08-02+03:23&entry.1917517419=2023-08-09+17:51&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
